components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -255,4 +253,13 @@ private String getAppID(String appName, String tenantDomain) throws IdentityOAut
             throw new IdentityOAuth2Exception("Error while getting application basic info.", e);
         }
     }
+
+    /**
+     * Validate grant type of access token.
+     *
+     * @param accessTokenDO access token to be switched
+     */
+    protected void validateGrantType(AccessTokenDO accessTokenDO) throws IdentityOAuth2Exception {
```

Comment:
can we block tokens with CC grant?. Otherwise developers will try to switch the cc tokens and might confuse why scopes didn't retrieved. Wdyt?